pkg/expr/sql/frame_table.go

```diff
@@ -90,7 +91,21 @@ func (ri *rowIter) Next(_ *mysql.Context) (mysql.Row, error) {
 		if field.NilAt(ri.row) {
 			continue
 		}
-		row[colIndex], _ = field.ConcreteAt(ri.row)
+		val, _ := field.ConcreteAt(ri.row)
+
+		// If the field is JSON, convert json.RawMessage to types.JSONDocument
+		if raw, ok := val.(json.RawMessage); ok {
+			doc, converted, err := types.JSON.Convert(raw)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert json.RawMessage to JSONDocument: %w", err)
+			}
+			if !converted {
+				return nil, fmt.Errorf("unexpected JSON conversion issue at row %d, column %s: value required type coercion", ri.row, ri.ft.Frame.Fields[colIndex].Name)
```

Comment:
Not sure I understand the error here. Meaning we were unable to convert?

Reply:
No, see the function signature for `types.JSON.Convert`

![image](https://github.com/user-attachments/assets/4f93d934-1081-49a6-9ce4-8e8249d7ff1f)

But I guess the variable name didn't help here. I'll rename to `inRange`